open-rmf/rmf-web
# Feature/time warning

## What's new

Adds warn time to task creation

![image](https://github.com/open-rmf/rmf-web/assets/5383623/59b47d05-73ea-4d8e-9d2c-5e7acd1073e0)

![image](https://github.com/open-rmf/rmf-web/assets/5383623/b8cd24b1-96db-4a72-9964-5b81b6f5200f)

![image](https://github.com/open-rmf/rmf-web/assets/5383623/9567dc56-b984-40e3-b0c2-762a820bae7a)

![image](https://github.com/open-rmf/rmf-web/assets/5383623/74734b2e-ab50-4ae1-bd85-fd1ab9497b56)

## How to test

* Run in office world
* select delivery task, with `Cart RFID` as `coke`, select pickup and dropoff
* check warn time, and select a warn time of before year 1970 (since we are using simulation time)
* submit
* a warning should immediately appear, that can be acknowledged or dismissed per normal
* acknowledging will append a log to the task log

## Self-checks

- [ ] I have prototyped this new feature (if necessary) on Figma 
- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test
